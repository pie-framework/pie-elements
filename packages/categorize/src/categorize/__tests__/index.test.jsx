import { shallow } from 'enzyme';
import React from 'react';
import { Categorize } from '../index';

jest.mock('@pie-lib/pie-toolbox/drag', () => ({
  uid: {
    withUid: jest.fn((a) => a),
    Provider: jest.fn((a) => a),
    generateId: jest.fn().mockReturnValue('1'),
  },
  withDragContext: jest.fn((n) => n),
}));

describe('categorize', () => {
  const defaultProps = {
    classes: {},
    session: {
      answers: [],
    },
    model: {
      choices: [],
      categories: [],
    },
  };
  let onAnswersChange;
  let onShowCorrectToggle;

  beforeEach(() => {
    onAnswersChange = jest.fn();
    onShowCorrectToggle = jest.fn();
  });
  const wrapper = (extras) => {
    const defaults = {
      ...defaultProps,
      onAnswersChange,
      onShowCorrectToggle,
    };
    const props = { ...defaults, ...extras };

    return shallow(<Categorize {...props} />);
  };

  describe('snapshots', () => {
    it('renders', () => {
      expect(wrapper()).toMatchSnapshot();
    });

    it('renders with feedback', () => {
      expect(
        wrapper({
          model: {
            ...defaultProps.model,
            correctness: 'correct',
            feedback: {
              correct: {
                type: 'default',
                default: 'Correct',
              },
              incorrect: {
                type: 'default',
                default: 'Incorrect',
              },
              partial: {
                type: 'default',
                default: 'Nearly',
              },
            },
          },
        }),
      ).toMatchSnapshot();
    });

    it('incorrect', () => {
      expect(wrapper({ incorrect: true })).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    describe('dropChoice', () => {
      it('calls onAnswersChange', () => {
        const w = wrapper();

        w.instance().dropChoice('1', { id: '1', choiceIndex: 0 });
        expect(onAnswersChange).toBeCalledWith([{ category: '1', choices: ['1'] }]);
      });
    });

    describe('removeChoice', () => {
      it('calls onAnswersChange', () => {
        const w = wrapper({
          session: { answers: [{ category: '1', choices: ['1'] }] },
        });

        w.instance().removeChoice({ id: '1', categoryId: '1', choiceIndex: 0 });

        expect(onAnswersChange).toBeCalledWith([{ category: '1', choices: [] }]);
      });
    });

    describe('showAnswers', () => {
      it('calls onShowCorrectToggle', () => {
        const w = wrapper({
          session: { answers: [{ category: '1', choices: ['1'] }] },
        });

        w.instance().toggleShowCorrect();

        expect(onShowCorrectToggle).toHaveBeenCalled();
      });
    });
  });
});
