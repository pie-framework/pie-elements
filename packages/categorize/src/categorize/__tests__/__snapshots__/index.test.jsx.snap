// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categorize snapshots incorrect 1`] = `
<div>
  <WithStyles(CorrectAnswerToggle)
    onToggle={[Function]}
    show={false}
    toggled={false}
  />
  <div
    style={
      Object {
        "flexDirection": "column-reverse",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
    </div>
    <WithStyles(Choices)
      choicePosition="above"
      choices={Array []}
      model={
        Object {
          "categories": Array [],
          "choices": Array [],
        }
      }
    />
  </div>
</div>
`;

exports[`categorize snapshots renders 1`] = `
<div>
  <WithStyles(CorrectAnswerToggle)
    onToggle={[Function]}
    show={false}
    toggled={false}
  />
  <div
    style={
      Object {
        "flexDirection": "column-reverse",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
    </div>
    <WithStyles(Choices)
      choicePosition="above"
      choices={Array []}
      model={
        Object {
          "categories": Array [],
          "choices": Array [],
        }
      }
    />
  </div>
</div>
`;

exports[`categorize snapshots renders with feedback 1`] = `
<div>
  <WithStyles(CorrectAnswerToggle)
    onToggle={[Function]}
    show={false}
    toggled={false}
  />
  <div
    style={
      Object {
        "flexDirection": "column-reverse",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctness": "correct",
            "feedback": Object {
              "correct": Object {
                "default": "Correct",
                "type": "default",
              },
              "incorrect": Object {
                "default": "Incorrect",
                "type": "default",
              },
              "partial": Object {
                "default": "Nearly",
                "type": "default",
              },
            },
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
    </div>
    <WithStyles(Choices)
      choicePosition="above"
      choices={Array []}
      model={
        Object {
          "categories": Array [],
          "choices": Array [],
          "correctness": "correct",
          "feedback": Object {
            "correct": Object {
              "default": "Correct",
              "type": "default",
            },
            "incorrect": Object {
              "default": "Incorrect",
              "type": "default",
            },
            "partial": Object {
              "default": "Nearly",
              "type": "default",
            },
          },
        }
      }
    />
  </div>
  <WithStyles(Feedback)
    correctness="correct"
    feedback={
      Object {
        "correct": Object {
          "default": "Correct",
          "type": "default",
        },
        "incorrect": Object {
          "default": "Incorrect",
          "type": "default",
        },
        "partial": Object {
          "default": "Nearly",
          "type": "default",
        },
      }
    }
  />
</div>
`;
